packages/core/lib/remote-parent-context.ts

```diff
@@ -0,0 +1,43 @@
+import { scaleProbabilityToMatchSamplingRate } from './sampler'
+import type { ParentContext } from './span'
+import { isParentContext, isSpanContext } from './validation'
+
+const TRACE_PERENT_REGEX = /^00-([0-9a-f]{32})-([0-9a-f]{16})-[0-9]{2}$/
+
+export default class RemoteParentContext implements ParentContext {
+  readonly id: string
+  readonly traceId: string
+
+  constructor (maybeParentContext: ParentContext | string, traceId?: string) {
+    if (isParentContext(maybeParentContext)) {
```

Comment:
Is this constructor confusing? Would it make more sense to only support `ParentContext` objects here?

```
new RemoteParentContext({ id: 'span-id', traceId: 'trace-id' })
```

Reply:
Yes, I think that might be significantly clearer.